Remove is_gamma_argument() function

phate opened the pull request:
This PR does the following:
1. Introduces the GammaArgument class, which represents a region argument in a gamma subregion.
2. Removes the is_gamma_argument() function.

haved commented on jlm/rvsdg/gamma.hpp:
```diff
@@ -465,6 +465,33 @@ inline gamma_node::gamma_node(jlm::rvsdg::output * predicate, size_t nalternativ
+  {}
+
+  static GammaArgument &
+  Create(rvsdg::region & region, structural_input & input, std::shared_ptr<const rvsdg::type> type)
```
Does the create method need a type specified, or can it get its type from the `input`?

phate replied:
Remove the type argument.

haved reviewed (approved):
Looks good, just one question

phate commented:
@haved Changes since last review:
1. Removed superfluous type from `Create()` method and `GammaArgument` constructor
2. Handed in a `GammaInput` instead of a `structural_input` to `GammaArgument`'s constructor and `Create()` method
